draft-pda-protocol.md

```diff
@@ -562,116 +594,264 @@ PA protocol. The structure of the leader share is similarly protocol specific:
-    case prio: PrioLeaderShare;
-    case hits: HitsLeaderShare;
+  select (PALeaderInputShare.proto) {
+    case prio: PrioLeaderInputShare;
+    case hits: HitsLeaderInputShare;
   };
-} PALeaderShare;
+} PALeaderInputShare;
 ~~~
 
-Note that the leader share is sent not encrypted.
+Note that only the helper's share is encrypted, and that the leader share is
+sent in plaintext.
 
 The leader responds to well-formed requests to `[leader]/upload_finish` with
 status 200 and an empty body. Malformed requests are handled as described in
 {{pa-error-common-aborts}}.
 
-[TODO: Since we're running the protocol with multiple cohorts of aggregators, the
-collector needs to decide how to pick which cohort has the "correct" output.
-This might be the cohort with the largest batch of inputs. Figure this out once
-the collection is specified.]
+## Collect {#pa-collect}
+
+~~~~
+Collector     Leader           Helper
+  |  collect 1  |                |
+  +------------->                |
+  |             |  aggregate 1   |
+  |             <---------------->
+  |             |  ...           |
+  |             |  aggregate L   |
+  |             <---------------->
+  |             |  output share  |
+  |             <---------------->
+  <-------------+                |
+  |  ...        |                |
+  |  collect N  |                |
+  +------------->                |
+  |             |  aggregate 1   |
+  |             <---------------->
+  |             |     ...        |
+  |             |  aggregate L   |
+  |             <---------------->
+  |             |  output share  |
+  |             <---------------->
+  <-------------+                |
+  v             v                v
+~~~~
+{: #pa-collect-flow title="Flow of the collect process with N collect requests
+and L aggregate requests per collect request."}
+
+[TODO: Decide how the collector's request is authenticated.]
+
+The collector interacts with the leader to produce the final aggregate output.
+This process consists of a sequence of *collect requests* issued to the leader.
+Before a request can succeed, the aggregators must have verified and aggregated
+enough reports and the leader must have obtained the helper's encrypted output
+share (see {{pa-aggregate}}). In general, the procedure by which the aggregators
+verify and aggregate reports depends on parameters carried by the collect
+request. This procedure is described below in {{pa-aggregate}}.
+
+### Collect Request
+
+The collect protocol is an iterative procedure driven by the collector and
+parameterized by a PAParam. At a high level, it proceeds as follows. First, the
+collector initializes local per-protocol state using the corresponding
+PAParam. This state object has the following member functions:
+
+- CreateRequest() -> PACollectReq: Creates a PACollectReq object, defined below,
+  that is sent to the leader to complete one iteration of the protocol.
+
+~~~
+struct {
+  PATask task;
+  Url helper;
+  PAProto proto;
```

Comment:
But there is no `PAParam` in this struct. I guess you could argue that the leader knows the `PAParam` since it has the task id. However, I tend to think that protocol messages should always be deserializaeable without extra context.